### PR TITLE
move dynamic settings metadata into static array

### PR DESCRIPTION
**Pull request by serprex**
avoids needing a migration when adding config,
originally put in database to available to ui through prisma,
but prisma is gone

### CODE REVIEW

File: nexus/catalog/migrations/V36__reduce_dynamic_settings.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+delete from dynamic_settings where config_value is null;
```
Comment (heavycrystal) [inline nexus/catalog/migrations/V36__reduce_dynamic_settings.sql:1]: comment on why this is done

**Review by heavycrystal — APPROVED** [review]

File: flow/peerdbenv/dynamicconf.go [hunk 2]
```diff
@@ -12,9 +12,98 @@ import (
 	"github.com/jackc/pgx/v5/pgtype"
 	"golang.org/x/exp/constraints"
 
+	"github.com/PeerDB-io/peer-flow/generated/protos"
 	"github.com/PeerDB-io/peer-flow/logger"
 )
 
+var DynamicSettings = [...]*protos.DynamicSetting{
+	{
+		Name: "PEERDB_MAX_SYNCS_PER_CDC_FLOW", DefaultValue: "32", ValueType: protos.DynconfValueType_UINT,
```
[the remark this answers is not in the document]
Comment (serprex) [reply flow/peerdbenv/dynamicconf.go:21]: no

File: flow/peerdbenv/dynamicconf.go [hunk 3]
```diff
@@ -23,23 +112,21 @@ func dynLookup(ctx context.Context, key string) (string, error) {
...
+	if err := conn.QueryRow(ctx, query, key).Scan(&value); err != nil && err != pgx.ErrNoRows {
 		logger.LoggerFromCtx(ctx).Error("Failed to get key", slog.Any("error", err))
 		return "", fmt.Errorf("failed to get key: %w", err)
 	}
 	if !value.Valid {
 		if val, ok := os.LookupEnv(key); ok {
 			return val, nil
 		}
-		return default_value.String, nil
+		for _, setting := range DynamicSettings {
```
[the remark this answers is not in the document]
Comment (serprex) [reply flow/peerdbenv/dynamicconf.go:124]: yes, but only 16 elements for now

File: nexus/catalog/migrations/V36__reduce_dynamic_settings.sql (hunk 1 above)
Comment (serprex) [reply nexus/catalog/migrations/V36__reduce_dynamic_settings.sql:1]: two lines down there's `set not null`

File: flow/peerdbenv/dynamicconf.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/peerdbenv/dynamicconf.go]: break even point is surprisingly low, good change

**Pull request merged**